src/trace/trace.cpp

```diff
@@ -119,46 +119,51 @@ Trace::Trace()
     registry_.create<otf2::definition::system_tree_node_domain>(
         system_tree_root_node_, otf2::common::system_tree_node_domain::shared_memory);
     const auto& sys = Topology::instance();
-    for (auto& package : sys.packages())
+    for (auto& cpu : sys.cpus())
```

Comment:
This is where not having the relations between package->core->cpu really hurts, as you constantly need to check whether things are there, making the code hard to understand in comparison. I'm not sure if I like that.